src/components/Reports/SubscriberHistory/SubscriberHistory.js

```diff
@@ -209,109 +210,118 @@ const SubscriberHistory = ({
-                                    {campaign.campaignName}
-                                  </a>
-                                )}
-                              </td>
-                              <td>{campaign.campaignSubject}</td>
-                              <td>
-                                <span
-                                  className={getDeliveryStatusCssClassName(campaign.deliveryStatus)}
-                                >
-                                  <FormattedMessage
-                                    id={
-                                      'subscriber_history.delivery_status.' +
-                                      campaign.deliveryStatus
-                                    }
-                                  />
-                                </span>
-                              </td>
-                              <td>{campaign.clicksCount}</td>
-                            </tr>
-                          ))}
-                        </>
-                      ) : (
-                        <tr>
-                          <td>
-                            <span className="bounceIn">
-                              <FormattedMessage id="subscriber_history.empty_data" />
-                            </span>
-                          </td>
-                          <td></td>
-                          <td></td>
-                          <td></td>
-                        </tr>
-                      )}
-                    </tbody>
-                  </table>
+                                    <FormattedMessage
+                                      id={
+                                        'subscriber_history.delivery_status.' +
+                                        campaign.deliveryStatus
+                                      }
+                                    />
+                                  </span>
+                                </td>
+                                <td>{campaign.clicksCount}</td>
+                              </tr>
+                            ))}
+                          </>
+                        ) : (
+                          <tr>
+                            <td>
+                              <span className="bounceIn">
+                                <FormattedMessage id="subscriber_history.empty_data" />
+                              </span>
+                            </td>
+                            <td></td>
+                            <td></td>
+                            <td></td>
+                          </tr>
+                        )}
+                      </tbody>
+                    </table>
+                  ) : (
+                    <SafeRedirect to="/Lists/MasterSubscriber/" />
+                  )}
                 </div>
               </div>
             </div>
           </div>
         </div>
       </section>
     </>
-  ) : (
-    <SafeRedirect to="/Lists/MasterSubscriber/" />
```

Comment:
It is fine moving the redirect inside so many nested divs?

Maybe you can move the redirect to the first lines and reduce the nesting. Please call me if I cannot express my idea.